### ISSUE TITLE
Allow populating admin password from environment variable

### ISSUE THREAD

**Issue by bdalpe** [criblio/helm-charts#75]
Allows for storing in Vault, etc. instead of Helm values

**Comment by Kavinraja-G** [comment]
@bdalpe -- Do we have anyother ways to pass securely the adminPassword, token and even license?
(looks like this is still in development?)

**Comment by bdalpe** [comment]
You are not required to populate those values to start the leader chart. You can set them after startup. They will be stored in config files on the PVC.

Long-term, we are looking to address this issue with a revision to the chart.

**Comment by Kavinraja-G** [comment]
Yeah that is doable but it defeats the purpose of automation and it will include manual intervention in my case (an user should login to the console and setup the configuration).

But I'm leaning towards the use of Envs to pass the credentials/licenses from source referencing to K8s secret/Vault.

**Comment by punkrokk** [comment]
@bdalpe why is this blocked ?

**Comment by bdalpe** [comment]
@punkrokk As of January, my internal request had not been implemented. I will look to do this with an initContainer in the future.

**Comment by druesendieb** [comment]
Any news here @bdalpe ?

**Comment by MrAlecJohnson** [comment]
I realise this is an old issue, but it would be hugely useful for our deployment. We're looking for a good way to securely automate licence, admin password and leader token, while managing a distributed setup using ArgoCD. 

**Licence**
It sounds like licence can be put in a file called `licenses.yml` in `$CRIBL_HOME/local/cribl` - according to [the licensing docs](https://docs.cribl.io/billing-licensing/on-prem-licensing/#licenses-in-airgapped-deployments).

**Leader token**
Am I right in thinking CRIBL_DIST_LEADER_URL/CRIBL_DIST_MASTER_URL can replace this? The [env var docs](https://docs.cribl.io/stream/environment-variables/#cribl_dist_leader_url) suggest it could work.

EDIT: this works, with these changes (assuming use of Hashicorp Vault): 
- in your worker values, set `config.useExistingSecret` to true
- in envValueFrom, include:
```
name: CRIBL_DIST_MASTER_URL
    valueFrom:
      secretKeyRef:
        name: <secret-name>
        key: <relevant-key>
```
- your secret should be in this form: `tcp://<leader-auth-token>@<leader's internal node balancer name>:4200?group=<workergroup-name>`
- note that the 'group' in your values file gets ignored, with the one in the secret used instead - but you still need to set a group in the values file because it's used to name other things in the chart

**Admin password**
I haven't yet found a workaround for this - anything I've missed? (It's also the one where automation is least vital - so could probably live without it)

(will update this once I've tested the licence and leader token workarounds above)
